privatevcs/validation/github_enterprise.go

```diff
@@ -3,13 +3,16 @@ package validation
 import (
 	"net/http"
 	"regexp"
+	"strings"
 )
 
 type githubEnterprisePattern struct {
 	Method string
 	Path   *regexp.Regexp
 }
 
+var tarballRegex *regexp.Regexp = regexp.MustCompile("^/(_?codeload/)?(?P<project>[^/]+/[^/]+)/legacy.tar.gz/[^/]+$")
```

Comment:
Maybe we could have just a couple of test cases to validate this regexp? I think it looks ok, but it started to get big so maybe we could test it just to be on the safe side 😄

Reply:
Yeah - makes sense. I've added tests for all the GHE requests. I haven't added tests for the other VCS providers just now because I don't want this task to explode.